src/app/wallet/proposals/add-proposal/add-proposal.component.ts

```diff
@@ -0,0 +1,151 @@
+      this.profileId = profile.id;
+    })
+
+    this.proposalFormGroup = this.formBuilder.group({
+      title: ['', Validators.compose([Validators.required, Validators.maxLength(50)])],
+      description: ['', Validators.compose([Validators.required, Validators.maxLength(2000)])],
+      options: this.formBuilder.array([
+        this.initOptionFields(),
+        this.initOptionFields()
+      ]),
+
+      // @TODO `nickname` and `email` use in the "Contact Information" section.
+      // nickname: ['', Validators.compose([Validators.maxLength(50)])],
+      // email: ['',  Validators.compose([Validators.maxLength(150)])]
+    });
+
+  }
+
+  initOptionFields(): FormGroup {
+    return this.formBuilder.group({
+      option: ['', Validators.compose([Validators.required, Validators.maxLength(50)])]
+    });
+  }
+
+  /**
+   * Programmatically generates a new option input field
+   *
+   * @public
+   * @method addNewInputField
+   * @return {none}
+   */
+  addNewInputField(): void {
+    const control = <FormArray>this.proposalFormGroup.controls.options;
+    control.push(this.initOptionFields());
+  }
+
+
+  /**
+   * Programmatically removes a recently generated option input field
+   *
+   * @public
+   * @method removeInputField
+   * @param i    {number}      The position of the object in the array that needs to removed
+   * @return {none}
+   */
+  removeInputField(i: number): void {
+    const control = <FormArray>this.proposalFormGroup.controls.options;
+    control.removeAt(i);
+  }
+
+  cancelAndDiscard() {
+    this.router.navigate(['/wallet/proposals']);
+  }
+
+  submitProposal() {
+    const dialogRef = this.dialog.open(ProposalConfirmationComponent);
+
+    dialogRef.componentInstance.setData({
+      ... this.proposalFormGroup.value,
+      options: this.proposalFormGroup.value.options.map(v => v.option),
+    },
+      (proposal) => this.addPost(proposal)
+    );
+  }
+
+  addPost(proposal: Proposal): void {
+
+    // check wallet status (unlock if locked ?).
+    this.modalService.unlock({timeout: 3}, () => this.addPostCall(proposal))
+  }
+
+  addPostCall(proposal: Proposal): void {
+    // get current block count.
+    this.peerService.getBlockCount().take(1).subscribe((startBlock: number) => {
+      /**
+       *  endBlockCount calculated based on formula <startBlockCount>.
+       *  endBlockCount = startBlockCount + 7 Days * 720 (particl block generate in a day).
+       */
+
+      const endBlockCount = startBlock + 7 * 720;
+      // add proposal.
+
+      this.proposalsService.post([
+        proposal.title,
+        proposal.description,
+        startBlock,
+        endBlockCount,
+        ...proposal.options
+      ]).subscribe((response) => {
+        this.snackbarService.open(
+          'Proposal posted successfully. Your proposal will get published shortly !',
+          'info'
+        )
+
+        // redirect to proposals page.
+        this.cancelAndDiscard();
```

Comment:
See comment above - we want to navigate somewhere else, not perform a cancel and/or discard action here - its somewhat confusing needing to use a cancelAndDsicard type of action here.